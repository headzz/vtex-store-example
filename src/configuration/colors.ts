const BLACK = '#000000'
const DARK_GREY = '#5B5B5B'
const GREEN_JUSSI = '#03FFA5'
const GREY_LVL1 = '#F2F2F2'
const PINK_JUSSI = '#EE0E7D'
const TRANSPARENT = 'transparent'
const WHITE = '#ffffff'

export default {
  BLACK,
  DARK_GREY,
  GREEN_JUSSI,
  GREY_LVL1,
  PINK_JUSSI,
  WHITE,
  TRANSPARENT
}
