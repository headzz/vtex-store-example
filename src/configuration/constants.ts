export const BLACK = '#000000'
export const WHITE = '#ffffff'
export const GREY_LVL1 = '#F2F2F2'
export const GREEN_JUSSI = '#03FFA5'
export const DARK_GREY = '#5B5B5B'
export const PINK_JUSSI = '#EE0E7D'
