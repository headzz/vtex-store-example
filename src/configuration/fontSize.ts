const SIZE_TEXT_SMALL = '1.2rem'
const SIZE_TEXT_NORMAL = '1.6rem'

const SIZE_TITLE_SMALL = '3.2rem'
const SIZE_TITLE_NORMAL = '6.4rem'

export default {
  SIZE_TEXT_SMALL,
  SIZE_TEXT_NORMAL,
  SIZE_TITLE_NORMAL,
  SIZE_TITLE_SMALL
}
