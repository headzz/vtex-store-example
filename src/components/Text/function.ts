import { fontSize } from 'configuration'

type FontSize = 'small' | 'normal' | ''

const textSize = {
  small: fontSize.SIZE_TEXT_SMALL,
  normal: fontSize.SIZE_TEXT_NORMAL
}

const titleSize = {
  normal: fontSize.SIZE_TITLE_NORMAL,
  small: fontSize.SIZE_TITLE_SMALL
}

/**
 * @function
 * @name checkFontSize
 * @param {FontSize} text defines the size for normal text.
 * @param {FontSize} title defines the size for title text.
 * @returns {string} size of desired text
 */

export const checkFontSize = (text: FontSize, title: FontSize): string => {
  if (title !== '') {
    return titleSize[title]
  }
  if (text !== '') {
    return textSize[text]
  }
  return textSize.normal
}
