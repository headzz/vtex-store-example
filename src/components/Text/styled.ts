import styled from 'styled-components'
import { checkFontSize } from './function'

type FontSize = 'small' | 'normal' | ''

type Font = {
  text: FontSize
  title: FontSize
  color?: string
  bold?: boolean
}

export const TextStyled = styled('p')<Font>`
  color: ${(props) => (props.color ? props.color : '')};
  font-size: ${({ text, title }) => checkFontSize(text, title)};
  font-weight: ${(props) => (props.bold ? '700' : '500')};
`
