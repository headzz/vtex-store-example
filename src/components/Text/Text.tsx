import { colors } from 'configuration'
import React from 'react'
import { TextStyled } from './styled'

type FontSize = 'small' | 'normal' | ''

type Font = {
  text: FontSize
  title: FontSize
  color?: string
  bold?: boolean
}

const defaultProps: Font = {
  bold: false,
  color: colors.BLACK,
  text: 'normal',
  title: ''
}

/**
 * @component
 * @name Text
 * @description creates text component to use on screen.
 * @namespace components components/Text
 * @param {FontSize} text Sets the size for text.
 * @param {FontSize} title Sets the size for title.
 * @param {element} children Component(s) that may appear inside the layout.
 * @param {string} color Specifies the desired color for the text.
 * @param {boolean} bold Sets the font weight to bold.
 *
 */
const Text: React.FC<Font> = ({ bold, children, color, text, title }) => {
  return (
    <TextStyled title={title} text={text} color={color} bold={bold}>
      {children}
    </TextStyled>
  )
}

Text.defaultProps = defaultProps
export default Text
