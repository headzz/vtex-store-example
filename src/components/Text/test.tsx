import { render, screen } from '@testing-library/react'
import { colors } from 'configuration'

import Text from './Text'

describe('Text', () => {
  it('should render component that small text with no color defined', () => {
    const { container } = render(
      <Text text="small" title="">
        Teste
      </Text>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyle('font-size: 1.2rem')
    expect(container.firstChild).toHaveStyle(`color: ${colors.BLACK}`)
    expect(screen.getByText('Teste')).toBeInTheDocument()
  })

  it('should render component that small text with no color defined', () => {
    const { container } = render(
      <Text text="small" title="">
        Teste
      </Text>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyle('font-size: 1.2rem')
    expect(container.firstChild).toHaveStyle(`color: ${colors.BLACK}`)
    expect(screen.getByText('Teste')).toBeInTheDocument()
  })

  it('should render small title', () => {
    const { container } = render(
      <Text text="" title="small">
        Teste
      </Text>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyle('font-size: 3.2rem')
    expect(container.firstChild).toHaveStyle(`color: ${colors.BLACK}`)
    expect(screen.getByText('Teste')).toBeInTheDocument()
  })

  it('should render normal title', () => {
    const { container } = render(
      <Text text="" title="normal">
        Teste
      </Text>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyle('font-size: 6.4rem')
    expect(container.firstChild).toHaveStyle(`color: ${colors.BLACK}`)
    expect(screen.getByText('Teste')).toBeInTheDocument()
  })

  it('should render component with bold text', () => {
    const { container } = render(
      <Text text="small" title="" bold>
        Teste
      </Text>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyle('font-size: 1.2rem')
    expect(container.firstChild).toHaveStyle('font-weight: 700')
    expect(container.firstChild).toHaveStyle(`color: ${colors.BLACK}`)
    expect(screen.getByText('Teste')).toBeInTheDocument()
  })

  it('should render text with green color', () => {
    const { container } = render(
      <Text text="" title="" color={colors.GREEN_JUSSI}>
        Teste
      </Text>
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveStyle('font-size: 1.6rem')
    expect(container.firstChild).toHaveStyle(`color: ${colors.GREEN_JUSSI}`)
    expect(screen.getByText('Teste')).toBeInTheDocument()
  })
})
