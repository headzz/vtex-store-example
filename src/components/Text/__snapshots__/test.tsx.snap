// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `Text should render component that small text with no color defined 1`
] = `
.c0 {
  color: #000000;
  font-size: 1.2rem;
  font-weight: 500;
}

<p
  class="c0"
  color="#000000"
  title=""
>
  Teste
</p>
`

exports[
  `Text should render component that small text with no color defined 2`
] = `
.c0 {
  color: #000000;
  font-size: 1.2rem;
  font-weight: 500;
}

<p
  class="c0"
  color="#000000"
  title=""
>
  Teste
</p>
`

exports[`Text should render component with bold text 1`] = `
.c0 {
  color: #000000;
  font-size: 1.2rem;
  font-weight: 700;
}

<p
  class="c0"
  color="#000000"
  title=""
>
  Teste
</p>
`

exports[`Text should render normal title 1`] = `
.c0 {
  color: #000000;
  font-size: 6.4rem;
  font-weight: 500;
}

<p
  class="c0"
  color="#000000"
  title="normal"
>
  Teste
</p>
`

exports[`Text should render small title 1`] = `
.c0 {
  color: #000000;
  font-size: 3.2rem;
  font-weight: 500;
}

<p
  class="c0"
  color="#000000"
  title="small"
>
  Teste
</p>
`

exports[`Text should render text with green color 1`] = `
.c0 {
  color: #03FFA5;
  font-size: 1.6rem;
  font-weight: 500;
}

<p
  class="c0"
  color="#03FFA5"
  title=""
>
  Teste
</p>
`
